services/business/istio_validations.go

```diff
@@ -21,31 +21,33 @@ func (in *IstioValidationsService) GetServiceValidations(namespace, service stri
 		return nil, err
 	}
 
-	objectCheckers := enabledCheckersFor(istioDetails)
+	objectCheckers := enabledCheckersFor(in, namespace, service, istioDetails)
 	objectCheckersCount := len(objectCheckers)
 
 	objectValidations := models.IstioValidations{}
-	validationsChannel := make(chan *models.IstioValidations, objectCheckersCount)
+	validationsChannels := make([]chan *models.IstioValidations, objectCheckersCount)
 
 	// Run checks for each IstioObject type
-	for _, objectChecker := range objectCheckers {
-		go func() {
-			validationsChannel <- objectChecker.Check()
-			close(validationsChannel)
-		}()
+	for i, objectChecker := range objectCheckers {
+		validationsChannels[i] = make(chan *models.IstioValidations)
+		go func(channel chan *models.IstioValidations, checker ObjectChecker) {
+			channel <- checker.Check()
+			close(channel)
+		}(validationsChannels[i], objectChecker)
 	}
 
 	// Receive validations and merge them into one object
-	for validation := range validationsChannel {
-		objectValidations.MergeValidations(validation)
+	for _, validation := range validationsChannels {
+		objectValidations.MergeValidations(<-validation)
 	}
 
 	// Get groupal validations for same kind istio objects
 	return objectValidations, nil
 }
 
-func enabledCheckersFor(istioDetails *kubernetes.IstioDetails) []ObjectChecker {
+func enabledCheckersFor(validationService *IstioValidationsService, namespace, service string, istioDetails *kubernetes.IstioDetails) []ObjectChecker {
 	return []ObjectChecker{
 		checkers.RouteRuleChecker{istioDetails.RouteRules},
+		&checkers.SidecarsChecker{Namespace: namespace, ServiceName: service, K8sClient: validationService.k8s},
```

Comment:
Here is the point. Adding this, you are coupling the object fetching to its validations. If you see the previous line, we pass the objects to checker. This way, we prevent the coupling.